pkg/xds/generator/proxy_template.go

```diff
@@ -126,4 +128,5 @@ func init() {
 
 func RegisterProfile(profileName string, generator ResourceGenerator) {
```

Comment:
`RegisteredProfile` is invoked with the gateway profile at some point via this line
https://github.com/kumahq/kuma/blob/d47398982190d5e48644c47648547ea647c5b344/pkg/plugins/runtime/gateway/plugin.go#L58